examples/chain-of-density/run.py

```diff
@@ -0,0 +1,228 @@
+import instructor
+import openai
+from pydantic import BaseModel, Field
+
+from pprint import pprint
+from pydantic import BaseModel, Field
+from typing import List, Dict
+
+
+class Summary(BaseModel):
+    """Represents a summary entry in the list.
+
+    Guidelines:
+        - The first summary should be long (4-5 sentences, ~80 words) yet highly non-specific,
+          containing little information beyond the entities marked as missing. Use overly verbose
+          language and fillers (e.g., "this article discusses") to reach ~80 words.
+        - Make every word count: rewrite the previous summary to improve flow and make space for
+          additional entities.
+        - Make space with fusion, compression, and removal of uninformative phrases like "the article discusses."
+        - The summaries should become highly dense and concise yet self-contained, i.e., easily understood
+          without the article.
+        - Missing entities can appear anywhere in the new summary.
+        - Never drop entities from the previous summary. If space cannot be made, add fewer new entities.
+    """
+
+    index: int = Field(..., description="Index of the summary in the chain.")
+    denser_summary: str = Field(..., description="Concise yet self-contained summary.")
+    included_entities: List[str] = Field(
+        ..., description="Correct list of Entities found in the summary."
+    )
+    missing_entities: List[str] = Field(
+        ...,
+        description="Correct list of Entities found absent from the summary that should be included in the next summary attempt.",
+    )
+
+
+# This multitask helper will be used to generate a chain of summaries.
+# Allows us to extract data via streaming to see resuls faster
+ChainOfDenseSummaries = instructor.MultiTask(
+    Summary,
+    name="chain-of-dense-summaries",
+    description="""
+        Repeat the following 2 steps 5 times.
```

Comment:
How good are models at following this sort of "counting instruction?"

Reply:
By adding that index it does ok lol